src/ctap/command.rs

```diff
@@ -134,19 +138,21 @@ impl TryFrom<cbor::Value> for AuthenticatorMakeCredentialParameters {
         let cred_param_vec = read_array(ok_or_missing(param_map.get(&cbor_unsigned!(4)))?)?;
         let mut pub_key_cred_params = vec![];
         for cred_param_map_value in cred_param_vec {
-            let cred_param_map = read_map(cred_param_map_value)?;
-            let cred_type = PublicKeyCredentialType::try_from(ok_or_missing(
-                cred_param_map.get(&cbor_text!("type")),
-            )?)?;
-            let alg = read_integer(ok_or_missing(cred_param_map.get(&cbor_text!("alg")))?)?;
-            pub_key_cred_params.push((cred_type, alg));
+            if let Ok(cred_param) = PublicKeyCredentialParameter::try_from(cred_param_map_value) {
+                pub_key_cred_params.push(cred_param);
+            }
         }
 
         let exclude_list = match param_map.get(&cbor_unsigned!(5)) {
             Some(entry) => {
                 let exclude_list_vec = read_array(entry)?;
                 let mut exclude_list = vec![];
                 for exclude_list_value in exclude_list_vec {
+                    if let Some(count) = MAX_CREDENTIAL_COUNT_IN_LIST {
+                        if exclude_list.len() as u64 >= count {
```

Comment:
`MAX_CREDENTIAL_COUNT_IN_LIST` should be a `usize` instead.

As far as I understand it, this parameter is local to this code, i.e. isn't used in any protocol, so a native size should be fine. Also, is there a use case where this would overflow an `u32`, to warrant using a `u64`? Unless there is more than 4GB of RAM, I don't see how that would ever happen.

Reply:
It is surfaced to the client in GetInfo. The Option is necessary since you can also decide to not filter at all.

Reply:
I didn't mean to remove the `Option` part of it. But representing as a `usize` internally and converting it to a `u64` in GetInfo would be simpler IMO.

Reply:
Done.